app/controllers/events_controller.rb

```diff
@@ -1,22 +1,66 @@
 class EventsController < ApplicationController
+
+  # require 'open-uri'
+
+  before_action :set_event, only: [:show, :edit, :update, :destroy]
+
   def index
+    #for the moment listing them all
+    @events = Event.all
   end
 
   def new
+    @event = Event.new
   end
 
   def create
+    @event = Event.new(event_params)
+    @event.user = current_user
+    if @event.save
+      redirect_to root_path
+    else
+      render :new
+    end
   end
 
   def show
+    @event = Event.find(params[:id])
   end
 
   def edit
+    @event = Event.find(params[:id])
   end
 
   def update
+    if @event.update(event_params)
+      redirect_to event_path(@event)
+    else
+      render :edit
+    end
   end
 
   def destroy
+    @event.destroy
```

Comment:
Do we want a redirect?

Reply:
Done. To root for the moment since it works fine